### PR TITLE
I created a feature branch, models, Dtos, controllers, repository and…

### PR DESCRIPTION
**Pull request by M-Jhay**
… service. Having issues creating Migrations.

### CODE REVIEW

File: src/DwitTech.NotificationService.Core/Dtos/EmailDto.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DwitTech.NotificationService.Core.Dtos
+{
+    public class EmailDto
+    {
+        [Required]
+        public string From { get; set; } = string.Empty;
```
Comment (dabensymon) [inline src/DwitTech.NotificationService.Core/Dtos/EmailDto.cs]: Do not default properties except its required. In this case, i dont believe defaulting is required.

File: src/DwitTech.NotificationService.Core/Interfaces/IEmailService.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DwitTech.NotificationService.Core.Interfaces
+{
+    public interface IEmailService
+    {
+        void SendEmail(EmailDto request);
```
Comment (dabensymon) [inline src/DwitTech.NotificationService.Core/Interfaces/IEmailService.cs]: Dont we want to know if sendEmail is successful?

File: src/DwitTech.NotificationService.Core/Services/EmailService.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            var email = new MimeMessage();
+            email.From.Add(MailboxAddress.Parse(request.From));
+            email.To.Add(MailboxAddress.Parse(request.To));
+            email.Bcc.Add(MailboxAddress.Parse(request.Bcc));
+            email.Cc.Add(MailboxAddress.Parse(request.Cc));
+            email.Subject = request.Subject;
+            email.Body = new TextPart(TextFormat.Html) { Text = request.Body };
+
+            using var smtp = new SmtpClient();
+            smtp.Connect("smtp.gmail.com",587,SecureSocketOptions.StartTls);
```
Comment (dabensymon) [inline src/DwitTech.NotificationService.Core/Services/EmailService.cs]: Do not hard code variables like this. "smtp.gmail.com" and 587 here should be system/global/environmental variables

Comment (dabensymon) [reply src/DwitTech.NotificationService.Core/Services/EmailService.cs]: By the way, this is beyond scope o!

File: src/DwitTech.NotificationService.Data/Entities/EmailEntity.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DwitTech.NotificationService.Data.Entities
+{
+    public class EmailEntity
+    {
+        [Key]
+        public int id { get; set; }
```
Comment (dabensymon) [inline src/DwitTech.NotificationService.Data/Entities/EmailEntity.cs]: Same as my initial comments. Shouldnt default properties except actually required

File: src/DwitTech.NotificationService.Data/Entities/EmailEntity.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DwitTech.NotificationService.Data.Entities
+{
+    public class EmailEntity
```
Comment (dabensymon) [inline src/DwitTech.NotificationService.Data/Entities/EmailEntity.cs]: This should be called Email, just Email

File: src/DwitTech.NotificationService.Data/Repository/EmailRepo.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DwitTech.NotificationService.Data.Repository
+{
+    public class EmailRepo : IEmailRepo
+    {
+        public IEnumerable<EmailEntity> Entities()
```
Comment (dabensymon) [inline src/DwitTech.NotificationService.Data/Repository/EmailRepo.cs]: Implementations are not currently required.

File: src/DwitTech.NotificationService.WebApi/Controllers/EmailController.cs [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        {
+            _emailService = emailService;
+        }
+
+        public IEmailService Get_emailService()
+        {
+            return _emailService;
+        }
+
+        /*[HttpGet]
```
Comment (dabensymon) [inline src/DwitTech.NotificationService.WebApi/Controllers/EmailController.cs]: Do not check in commented out code

File: src/DwitTech.NotificationService.Core/Dtos/EmailDto.cs (hunk 1 above)
Comment (M-Jhay) [reply src/DwitTech.NotificationService.Core/Dtos/EmailDto.cs]: > Do not default properties except its required. In this case, i dont believe defaulting is required.

Resolved

File: src/DwitTech.NotificationService.Core/Interfaces/IEmailService.cs (hunk 2 above)
Comment (M-Jhay) [reply src/DwitTech.NotificationService.Core/Interfaces/IEmailService.cs]: > 



> Dont we want to know if sendEmail is successful?

Resolved

File: src/DwitTech.NotificationService.Core/Services/EmailService.cs (hunk 3 above)
Comment (M-Jhay) [reply src/DwitTech.NotificationService.Core/Services/EmailService.cs]: > Do not hard code variables like this. "smtp.gmail.com" and 587 here should be system/global/environmental variables

Resolved

Comment (M-Jhay) [reply src/DwitTech.NotificationService.Core/Services/EmailService.cs]: > By the way, this is beyond scope o!

Resolved

File: src/DwitTech.NotificationService.Data/Entities/EmailEntity.cs (hunk 4 above)
Comment (M-Jhay) [reply src/DwitTech.NotificationService.Data/Entities/EmailEntity.cs]: > Same as my initial comments. Shouldnt default properties except actually required

Resolved

File: src/DwitTech.NotificationService.Data/Entities/EmailEntity.cs (hunk 5 above)
Comment (M-Jhay) [reply src/DwitTech.NotificationService.Data/Entities/EmailEntity.cs]: > This should be called Email, just Email

Resolved

File: src/DwitTech.NotificationService.Data/Repository/EmailRepo.cs (hunk 6 above)
Comment (M-Jhay) [reply src/DwitTech.NotificationService.Data/Repository/EmailRepo.cs]: > Implementations are not currently required.

Resolved

File: src/DwitTech.NotificationService.WebApi/Controllers/EmailController.cs (hunk 7 above)
Comment (M-Jhay) [reply src/DwitTech.NotificationService.WebApi/Controllers/EmailController.cs]: > Do not check in commented out code

Resolved

File: src/DwitTech.NotificationService.Data/Entities/Email.cs [hunk 8]
```diff
@@ -25,5 +25,7 @@ public class Email
         public string Subject { get; set; }
 
         public string Body { get; set; }
+
+        public EmailStatus Status { get; set; }
```
Comment (johnny2565) [inline src/DwitTech.NotificationService.Data/Entities/Email.cs:29]: The pull request is okay by me

**Review by giftedGeorge — COMMENTED** [review]
Looks good to me boss.

File: src/DwitTech.NotificationService.Data/Entities/Email.cs [hunk 9]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DwitTech.NotificationService.Data.Entities
+{
+    public class Email
```
Comment (dabensymon) [inline src/DwitTech.NotificationService.Data/Entities/Email.cs]: This should inherit from the BaseEntity

File: tests/DwitTech.NotificationService.WebApi.Tests/DwitTech.NotificationService.WebApi.Tests.csproj [hunk 10]
```diff
@@ -9,6 +9,11 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="MailKit" Version="3.5.0" />
```
Comment (dabensymon) [inline tests/DwitTech.NotificationService.WebApi.Tests/DwitTech.NotificationService.WebApi.Tests.csproj]: what is this MailKit everywhere here for? If not in use, please remove it.

File: src/DwitTech.NotificationService.Data/Repository/IEmailRepo.cs [hunk 11]
```diff
@@ -0,0 +1,14 @@
+﻿using DwitTech.NotificationService.Data.Entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DwitTech.NotificationService.Data.Repository
+{
+    internal interface IEmailRepo
```
Comment (dabensymon) [inline src/DwitTech.NotificationService.Data/Repository/IEmailRepo.cs]: This shouldn't be internal.

File: src/DwitTech.NotificationService.Core/Dtos/EmailDto.cs [hunk 12]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DwitTech.NotificationService.Core.Dtos
+{
+    public class EmailDto
```
Comment (dabensymon) [inline src/DwitTech.NotificationService.Core/Dtos/EmailDto.cs:10]: This should inherit from BaseDto

File: src/DwitTech.NotificationService.Data/Entities/Email.cs (hunk 9 above)
Comment (M-Jhay) [reply src/DwitTech.NotificationService.Data/Entities/Email.cs]: > 

Resolved

File: tests/DwitTech.NotificationService.WebApi.Tests/DwitTech.NotificationService.WebApi.Tests.csproj (hunk 10 above)
Comment (M-Jhay) [reply tests/DwitTech.NotificationService.WebApi.Tests/DwitTech.NotificationService.WebApi.Tests.csproj]: > what is this MailKit everywhere here for? If not in use, please remove it.

Resolved

File: src/DwitTech.NotificationService.Data/Entities/Email.cs (hunk 9 above)
Comment (M-Jhay) [reply src/DwitTech.NotificationService.Data/Entities/Email.cs]: > This should inherit from the BaseEntity

Resolved

File: src/DwitTech.NotificationService.Data/Repository/IEmailRepo.cs (hunk 11 above)
Comment (M-Jhay) [reply src/DwitTech.NotificationService.Data/Repository/IEmailRepo.cs]: > This shouldn't be internal.

Resolved

File: src/DwitTech.NotificationService.Data/Entities/Email.cs [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DwitTech.NotificationService.Data.Entities
+{
+    public class Email : BaseEntity
+    {
+        public int Id { get; set; }
```
Comment (dabensymon) [inline src/DwitTech.NotificationService.Data/Entities/Email.cs]: I believe the BaseEntity already defines an Id so this field shouldnt be here.

File: src/DwitTech.NotificationService.Data/Context/NotificationDbContext.cs [hunk 14]
```diff
@@ -8,15 +8,16 @@
...
 
 namespace DwitTech.NotificationService.Data.Context
 {
     public class NotificationDbContext : DbContext
     {
-        public NotificationDbContext(DbContextOptions options) : base(options)
+        public NotificationDbContext(DbContextOptions<NotificationDbContext> options) : base(options)
         {
         }
+        public DbSet<Email> Entities { get; set; }
```
Comment (dabensymon) [inline src/DwitTech.NotificationService.Data/Context/NotificationDbContext.cs]: This should be Emails not Entities

**Review by dabensymon — CHANGES_REQUESTED** [review]
Looks good apart from the 2 minor comments

File: src/DwitTech.NotificationService.Data/Entities/Email.cs (hunk 13 above)
Comment (M-Jhay) [reply src/DwitTech.NotificationService.Data/Entities/Email.cs]: > I believe the BaseEntity already defines an Id so this field shouldnt be here.

Resolved

File: src/DwitTech.NotificationService.Data/Context/NotificationDbContext.cs (hunk 14 above)
Comment (M-Jhay) [reply src/DwitTech.NotificationService.Data/Context/NotificationDbContext.cs]: > This should be Emails not Entities

Resolved

**Review by dabensymon — APPROVED** [review]
Good job

**Pull request merged**